framework/src/base/Assembly.C

```diff
@@ -4745,42 +4754,55 @@ Assembly::havePRefinement()
       // perhaps silly helpers
       if (!user_added_helper_type)
       {
-        auto & fe_container_dim = fe_container[dim];
+        auto & fe_container_dim = libmesh_map_find(fe_container, dim);
         auto fe_it = fe_container_dim.find(helper_type);
         mooseAssert(fe_it != fe_container_dim.end(), "We should have the helper type");
         delete fe_it->second;
         fe_container_dim.erase(fe_it);
       }
+      else if (user_disabled_lagrange_p_refinement)
+        // We cannot invalidate FE references so we can't use the logic in the above if block
+        for (const auto & lagrange_type : lagrange_types)
+        {
+          auto & fe_container_dim = libmesh_map_find(fe_container, dim);
+          auto fe_it = fe_container_dim.find(lagrange_type);
+          if (lagrange_type == helper_type)
+            mooseAssert(fe_it != fe_container_dim.end(), "We should have the helper type");
```

Comment:
never seen that

how does it compile in opt? do mooseAssert completely disapeear (thus problem), or do they turn into a neutral statement

Reply:
```
#ifdef NDEBUG
#define mooseAssert(asserted, msg) ((void)0)
#else
```

Reply:
thx